source/common/filesystem/inotify/watcher_impl.cc

```diff
@@ -40,7 +40,8 @@ void WatcherImpl::addWatch(const std::string& path, uint32_t events, OnChangedCb
   std::string directory = last_slash != 0 ? path.substr(0, last_slash) : "/";
   std::string file = StringUtil::subspan(path, last_slash + 1, path.size());
 
-  int watch_fd = inotify_add_watch(inotify_fd_, directory.c_str(), IN_ALL_EVENTS);
+  const uint32_t watch_mask = IN_MODIFY | IN_MOVED_TO;
+  int watch_fd = inotify_add_watch(inotify_fd_, directory.c_str(), watch_mask);
```

Comment:
I'm happy to submit this change separately. It's indirectly related to the rest of this PR.

Reply:
I think I'm OK with this in this PR -- it seems related enough, but does it make sense to add a test that would fail if you don't have this change? Like a change that proves our mock did not get called for some event mask that is not IN_MODIFY or IN_MOVED_TO.

Also it looks you only tested IN_MODIFY; also add a test for IN_MOVED_TO?

Reply:
I think your first suggestion is a good one, but given how the `Watcher` implementations are abstracted, I don't think we have a good way to author that kind of test. Watcher callbacks already do their own event masking (based on the `Watcher::Events` values) so to the best we can test is that some inotify event that _isn't_ in the `inotify_add_watch()` mask doesn't also trigger one of our `Watcher::Events`-masked events and run our callback, and that could only happen if our callback-level `Watcher::Events` masking code was incorrect (by categorizing an inotify event as the wrong `Watcher::Events` value). Did I say that in a way that makes sense? :wink:

Also, the existing tests were all covering the `Watcher::Events::MovedTo` / `IN_MOVED_TO` behavior, so I think that's still well-covered. Or am I missing an inotify-specific test that you had in mind?